druid/src/widget/slider.rs

```diff
@@ -68,7 +88,25 @@ impl Slider {
         let scalar = ((mouse_x + self.x_offset - knob_width / 2.) / (slider_width - knob_width))
             .max(0.0)
             .min(1.0);
-        self.min + scalar * (self.max - self.min)
+        let mut value = self.min + scalar * (self.max - self.min);
+        if let Some(step) = self.step {
+            let max_step_value =
+                (((value - self.min) / step).floor() * step + self.min).min(self.max);
+            if value > max_step_value {
+                // edge case: make sure max is reachable
+                let left_dist = value - max_step_value;
+                let right_dist = self.max - value;
+                value = if left_dist < right_dist {
+                    max_step_value
+                } else {
+                    self.max
+                };
+            } else {
+                // snap to discrete intervals
+                value = (((value - self.min) / step).round() * step + self.min).min(self.max);
```

Comment:
same i don't think `.min(self.max)` is needed because `value < max_step_value` and at most round to `max_step_value` which is less than `self.max`

Reply:
> value < max_step_value

That is actually not true, since `value` could very well be `self.max` here.
I updated the code to remove the call for in the other location though.